platform/linux-generic/pktio/dpdk.c

```diff
@@ -910,23 +901,19 @@ static inline int mbuf_to_pkt_zero(pktio_entry_t *pktio_entry,
 
 		data = rte_pktmbuf_mtod(mbuf, char *);
 		pkt_len = rte_pktmbuf_pkt_len(mbuf);
-
 		pkt_hdr = pkt_hdr_from_mbuf(mbuf);
 
+		packet_parse_reset(&parsed_hdr, 1);
+		packet_set_len(&parsed_hdr, pkt_len);
```

Comment:
This function needs to be optimized for the case where parsing is not required. In the current form there is ~20% performance drop with odp_l2fwd example (`odp_l2fwd -a 0 -i 0,1 -m 0 -c 1`). On a quick glance `packet_parse_reset()`, `packet_set_len`,  and `_odp_packet_copy_cls_md()` are not required when parsing is not enabled.

Reply:
v4.